### PR TITLE
Generic Processors

### PR DESCRIPTION
**Pull request by boulter**
I decided to just implement a more generic version of https://github.com/HubSpot/jinjava/pull/1044 so we could easily add processors for pre and post execution of nodes and now filters. In the future we could add tags and functions to the builder.

### CODE REVIEW

**Review by jasmith-hs — CHANGES_REQUESTED** [review]
I like the idea of making this usable across more operations in Jinjava. Are there any filters, in particular, you're interested in timing?

**Comment by boulter** [comment]
> I like the idea of making this usable across more operations in Jinjava. Are there any filters, in particular, you're interested in timing?

I'm not really concerned with filters right now, but it was an easy demonstration for how to use processors for functionality besides nodes.

File: src/main/java/com/hubspot/jinjava/lib/filter/Filter.java [hunk 1]
```diff
@@ -67,7 +68,18 @@ default Object filter(
...
+    try {
+      if (processors != null && processors.getFilterPreProcessor() != null) {
+        processors.getFilterPreProcessor().accept(this, interpreter);
+      }
+      return filter(var, interpreter, filterArgs);
+    } finally {
+      if (processors != null && processors.getFilterPostProcessor() != null) {
+        processors.getFilterPostProcessor().accept(this, interpreter);
+      }
+    }
```
[the remark this answers is not in the document]
Comment (boulter) [reply src/main/java/com/hubspot/jinjava/lib/filter/Filter.java]: yeah, I know it won't catch everything, but it does catch most. I added a comment to explain that.

File: src/main/java/com/hubspot/jinjava/JinjavaConfig.java [hunk 2]
```diff
@@ -480,13 +495,22 @@ public Builder withObjectUnwrapper(ObjectUnwrapper objectUnwrapper) {
       return this;
     }
 
+    @Deprecated
+    /**
+     * @deprecated use {@link #withProcessors(JinjavaProcessors)}
+     */
```
[the remark this answers is not in the document]
Comment (boulter) [reply src/main/java/com/hubspot/jinjava/JinjavaConfig.java:494]: I did like the idea of not storing the `nodePreProcessor`, so I removed it in favor of building on the existing `Processors` with a new builder method.

File: src/main/java/com/hubspot/jinjava/lib/filter/Filter.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jasmith-hs) [reply src/main/java/com/hubspot/jinjava/lib/filter/Filter.java]: You could instead add processors within `JinjavaInterpreterResolver.invoke` or some other place within the ELResolver code to get around this.

**Comment by boulter** [comment]
I've removed the filter processors for now since I don't need them.

File: src/main/java/com/hubspot/jinjava/lib/filter/Filter.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (tes966) [reply src/main/java/com/hubspot/jinjava/lib/filter/Filter.java]: https://github.com/HubSpot/jinjava/pull/1045?new_signup=true?new_signup=true#discussion_r1161054773

**Pull request merged**